plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergLocalConcurrentWrites.java

```diff
@@ -657,6 +657,64 @@ void testConcurrentUpdateAndInserts()
         }
     }
 
+    @Test
+    public void testConcurrentMerge()
+            throws Exception
+    {
+        int threads = 3;
+        CyclicBarrier barrier = new CyclicBarrier(threads);
+        ExecutorService executor = newFixedThreadPool(threads);
+        String tableName = "test_concurrent_merges_table_" + randomNameSuffix();
+
+        assertUpdate("CREATE TABLE " + tableName + " (a, part)  WITH (partitioning = ARRAY['part']) AS VALUES (1, 10), (11, 20), (21, 30), (31, 40)", 4);
+        // Add more files in the partition 30
+        assertUpdate("INSERT INTO " + tableName + " VALUES (22, 30)", 1);
+
+        try {
+            // merge data concurrently by using non-overlapping partition predicate
+            executor.invokeAll(ImmutableList.<Callable<Void>>builder()
+                            .add(() -> {
+                                barrier.await(10, SECONDS);
+                                getQueryRunner().execute(
+                                        """
+                                                MERGE INTO %s t USING (VALUES (12, 20)) AS s(a, part)
+                                                  ON (FALSE)
+                                                    WHEN NOT MATCHED THEN INSERT (a, part) VALUES(s.a, s.part)
+                                                """.formatted(tableName));
```

Comment:
Could you remove redundant spaces likes https://github.com/trinodb/trino/pull/23958?